subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -367,9 +367,36 @@ int fota_download_init(fota_download_callback_t client_callback)
 		return err;
 	}
 
+	first_fragment = true;
 	return 0;
 }
 
+int fota_download_cancel(void)
+{
+	int err = 0;
+
+	if (dlc.fd == -1) {
+		/* Download not started, aborted or completed */
+		err = -EALREADY;
+		LOG_WRN("%s invalid state", __func__);
+	} else {
+		err = download_client_disconnect(&dlc);
+		if (err) {
+			LOG_ERR("%s failed to disconnect: %d", __func__, err);
+			return err;
+		}
```

Comment:
Please add a blank line after closing bracket, it's a bit cramped.

Reply:
Added